wrapper.py

```diff
@@ -0,0 +1,1278 @@
+            raise NotImplementedError
+        return torch.prod(a, dtype=dtype, out=out)
+    elif is_sequence(axis):
+        raise NotImplementedError
+    return torch.prod(a, axis=axis, dtype=dtype, keepdims=keepdims, out=out)
+
+
+def corrcoef(x, y=None, rowvar=True, bias=NoValue, ddof=NoValue, *, dtype=None):
+    if bias is not None or ddof is not None:
+        # deprecated in NumPy
+        raise NotImplementedError
+    if rowvar is False:
+        x = x.T
+    if y is not None:
+        raise NotImplementedError
+    if dtype is not None:
+        x = x.type(dtype)
+    return torch.corrcoef(x)
+
+
+def concatenate(ar_tuple, axis=0, out=None, dtype=None, casting="same_kind"):
+    if casting != "same_kind":
+        raise NotImplementedError   # XXX
+    if dtype is not None:
+        # XXX: map numpy dtypes
+        ar_tuple = tuple(ar.type(dtype) for ar in ar_typle)
+    return torch.cat(ar_tuple, axis, out=out)
+
+
+def squeeze(a, axis=None):
+    if axis is None:
+        return torch.squeeze(a)
+    else:
+        return torch.squeeze(a, axis)
+
+
+def bincount(x, /, weights=None, minlength=0):
+    return torch.bincount(x, weights, minlength)
+
+
+###### module-level queries of object properties
+
+def ndim(a):
+    return torch.as_tensor(a).ndim
+
+
+def shape(a):
+    return tuple(torch.as_tensor(a).shape)
+
+
+def size(a, axis=None):
+    if axis is None:
+        return torch.as_tensor(a).numel()
+    else:
+        return torch.as_tensor(a).shape[axis]
+
+
+###### shape manipulations and indexing
+
+def reshape(a, newshape, order='C'):
+    if order != 'C':
+        raise NotImplementedError
+    return torch.reshape(a, newshape)
+
+
+def broadcast_to(array, shape, subok=False):
+    if subok:
+        raise NotImplementedError
+    return torch.broadcast_to(array, shape)
+
+
+from torch import broadcast_shapes
+
+
+def broadcast_arrays(*args, subok=False):
+    if subok:
+        raise NotImplementedError
+    return torch.broadcast_tensors(*args)
+
+
+def unravel_index(indices, shape, order='C'):
+# cf https://github.com/pytorch/pytorch/pull/66687
+# this version is from 
+# https://discuss.pytorch.org/t/how-to-do-a-unravel-index-in-pytorch-just-like-in-numpy/12987/3
+    if order != 'C':
+        raise NotImplementedError
+    result = []
+    for index in indices:
+        out = []
+        for dim in reversed(shape):
+            out.append(index % dim)
+            index = index // dim
+        result.append(tuple(reversed(out)))
+    return result
+
+
+def ravel_multi_index(multi_index, dims, mode='raise', order='C'):
+    # XXX: not available in pytorch, implement
+    return sum(idx*dim for idx, dim in zip(multi_index, dims))
+
+
+
+###### reductions
+
+def argmax(a, axis=None, out=None, *, keepdims=NoValue):
+    if axis is None:
+        result = torch.argmax(a, keepdims=bool(keepdims))
+    else:
+        result = torch.argmax(a, axis, keepdims=bool(keepdims))
+    if out is not None:
+        out[...] = result
```

Comment:
```suggestion
        out.copy_(result)
```

Reply:
Also, how are we going to do with checks (preconditions and so). Probably we don't want to care too much about them for now, but otherwise probably you need to check something like `out.shape == result.shape` or so.

This is the kind of argument that we implemented as a decorator in PrimTorch.